app/views/pagy/_next_page.html.erb

```diff
@@ -1,12 +1,12 @@
 <li class="pub-c-pagination__item pub-c-pagination__item--next">
-  <%= link_to url, rel: 'next', remote: remote, class: "pub-c-pagination__link govuk-link" do %>
+  <%= link_to "#{request.path}?page=#{pagy.next}", rel: 'next', class: "pub-c-pagination__link govuk-link" do %>
```

Comment:
this is rudimentary but works for our use case. I tried a couple of other things but would need a fair bit of work to deal with all the cases